src/ZODB/mvccadapter.py

```diff
@@ -27,10 +27,9 @@ def __init__(self, storage):
 
     def __getattr__(self, name):
         if name in self._copy_methods:
-            if hasattr(self._storage, name):
-                m = getattr(self._storage, name)
-                setattr(self, name, m)
-                return m
+            m = getattr(self._storage, name)
```

Comment:
For the record, I note that I think this is nearly essentially equivalent. Both previous `if` statements, if false, will fall through to the end of the method, which raises `AttributeError`. So the `getattr` here will either raise `AttributeError` (with generally a more helpful error message), or it may raise some other exception that would previously have been masked by `hasattr` in Python 2 (extremely rarely).